### PR TITLE
refactor: remove content and image from note

### LINKED ISSUE #156: Notes: refactor note class to remove unused fields

### PR DESCRIPTION
**Pull request by santiagoarayaramos**
# Description

Removed unused `content` and `image` attributes from note. For this, I also updated NotesToFlashcard to use the markdown file instead of note content and fixed the tests to work with this new method.

Apart from this, I refactor EditNote and EditMarkdown. Currently, we don't check if the selected note is null when we are editing it or its markdown. Instead, we give default values to the functions that need it, so in case of error where no note is selected (which should not happen) we won't know what's wrong, or even detect an error.

Instead of this, I proposed showing an error screen if this unexpected error should happen, which allows the user to go back (like what is done if no user is found in EditNoteScreen), and then if not then we assume (`!!`) that selected note is correct (we still need to use !! since `note` / `selectedNote` is an externally mutable variable so can smart-cast).

I also implemented other smaller style changes (documentation, remove unnecessary comments, etc...), to improve code readability and future maintenance.

Fixes #156

# How Has This Been Tested?

Fixed existing tests to work with new refactored code

  
# Dependencies

No new dependencies

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged in downstream modules, no conflicts with main

### CODE REVIEW

File: app/src/main/java/com/github/onlynotesswent/ui/overview/EditNote.kt [hunk 1]
```diff
@@ -367,78 +357,20 @@ fun PdfCard(
...
 
-/** Displays a screen indicating that the user was not found. */
+/**
+ * Displays a screen indicating that the user was not found.
+ *
+ * @param errorText The error message to display.
+ */
 @Composable
-fun UserNotFoundEditScreen() {
+fun ErrorScreen(errorText: String) {
```
Comment (Roshan-y) [inline app/src/main/java/com/github/onlynotesswent/ui/overview/EditNote.kt:366]: Nice idea making a common error screen, we could probably make this common composable for the whole app and also add a TAG parameter

File: app/src/main/java/com/github/onlynotesswent/utils/NotesToFlashcard.kt [hunk 2]
```diff
@@ -28,35 +31,22 @@ class NotesToFlashcard(
...
-      // TODO: modify note.content to markdown format when we implement markdown support
-      openAIClient.sendRequest(
-          promptPrefix + note.content,
-          { parseFlashcardsFromJson(it, note, onSuccess, onFailure) },
-          { onFailure(it) })
-    } catch (e: Exception) {
-      Log.e(TAG, "Unexpected error in convertNoteToFlashcards", e)
-      onFailure(e)
-    }
+    // Download the note text file,
```
Comment (Roshan-y) [inline app/src/main/java/com/github/onlynotesswent/utils/NotesToFlashcard.kt:37]: Nice job removing redundant try catch

File: app/src/test/java/com/github/onlynotesswent/model/authentication/GoogleCredSignInTest.kt
Comment (Roshan-y) [inline app/src/test/java/com/github/onlynotesswent/model/authentication/GoogleCredSignInTest.kt:1]: Nice refactoring but it is a bit off topic for this pr

File: app/src/test/java/com/github/onlynotesswent/utils/NotesToFlashcardTest.kt [hunk 3]
```diff
@@ -92,14 +99,26 @@ class NotesToFlashcardTest {
 
     // Mock FlashcardRepository and set up FlashcardViewModel with it
     flashcardViewModel = FlashcardViewModel(mockFlashcardRepository)
-    notesToFlashcard = NotesToFlashcard(flashcardViewModel, mockFileViewModel, mockOpenAI)
+    notesToFlashcard =
+        NotesToFlashcard(flashcardViewModel, mockFileViewModel, mockOpenAI, mockContext)
+
+    val testFile = File.createTempFile("test", ".md")
```
Comment (Roshan-y) [inline app/src/test/java/com/github/onlynotesswent/utils/NotesToFlashcardTest.kt:105]: Nice job making a test file

**Review by Roshan-y — CHANGES_REQUESTED** [review]
Overall very well and carefully refactored, making sure every file involved with note was refactored. Using `!!` and changing it from `?:` was also a good practice everyone should follow. Two small things, first of all the GoogleCredSignInTest refactor was a bit out of context for this pr. Second of all I think it might be a good idea to make the ErrorScreen composable a common composable with also the tag as a parameter to use in other screens.

File: app/src/test/java/com/github/onlynotesswent/utils/NotesToFlashcardTest.kt [hunk 4]
```diff
@@ -25,12 +27,14 @@ import org.mockito.Mockito.times
...
 import org.mockito.MockitoAnnotations
+import org.mockito.kotlin.any
+import org.mockito.kotlin.eq
 import org.robolectric.RobolectricTestRunner
 
 @RunWith(RobolectricTestRunner::class)
 class NotesToFlashcardTest {
   // Function to allow null arguments for Mockito when needed
-  private fun <T> any(): T = Mockito.any()
+  private fun <T> nullableAny(): T = Mockito.any()
```
[the remark this answers is not in the document]
Comment (santiagoarayaramos) [reply app/src/test/java/com/github/onlynotesswent/utils/NotesToFlashcardTest.kt:37]: Yes, but now I replaced it with mockito kotlin's anyOrNull which is already defined

File: app/src/main/java/com/github/onlynotesswent/ui/overview/EditNote.kt (hunk 1 above)
Comment (santiagoarayaramos) [reply app/src/main/java/com/github/onlynotesswent/ui/overview/EditNote.kt:366]: Totally agree, I'll leave this for a future PR where we can create utility classes for UI, tests, MVVM, which will make coding them more uniform and reduce redundant code

File: app/src/test/java/com/github/onlynotesswent/model/authentication/GoogleCredSignInTest.kt
Comment (santiagoarayaramos) [reply app/src/test/java/com/github/onlynotesswent/model/authentication/GoogleCredSignInTest.kt:1]: You're right, just spotted it so I preferred to remove it before forgetting

File: app/src/test/java/com/github/onlynotesswent/utils/NotesToFlashcardTest.kt [hunk 5]
```diff
@@ -18,20 +20,19 @@ import org.junit.runner.RunWith
...
 import org.mockito.Mock
-import org.mockito.Mockito
 import org.mockito.Mockito.anyString
 import org.mockito.Mockito.doAnswer
 import org.mockito.Mockito.times
 import org.mockito.Mockito.verify
 import org.mockito.Mockito.`when`
 import org.mockito.MockitoAnnotations
+import org.mockito.kotlin.any
+import org.mockito.kotlin.anyOrNull
```
Comment (Roshan-y) [inline app/src/test/java/com/github/onlynotesswent/utils/NotesToFlashcardTest.kt:30]: I had no idea this existed I have been creating my own fun, so thank you for that

File: app/src/test/java/com/github/onlynotesswent/utils/NotesToFlashcardTest.kt (hunk 4 above)
[the remark this answers is not in the document]
Comment (Roshan-y) [reply app/src/test/java/com/github/onlynotesswent/utils/NotesToFlashcardTest.kt]: Nice

File: app/src/main/java/com/github/onlynotesswent/ui/overview/EditNote.kt (hunk 1 above)
Comment (Roshan-y) [reply app/src/main/java/com/github/onlynotesswent/ui/overview/EditNote.kt:366]: Ok perfect

File: app/src/test/java/com/github/onlynotesswent/model/authentication/GoogleCredSignInTest.kt
Comment (Roshan-y) [reply app/src/test/java/com/github/onlynotesswent/model/authentication/GoogleCredSignInTest.kt:1]: Okok

**Review by Roshan-y — APPROVED** [review]
Yup all good now good job!

**Pull request merged**